substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -997,6 +998,11 @@ private static void onClassFileLoadHook(@SuppressWarnings("unused") JvmtiEnv jvm
         if (loader.equal(nullHandle())) { // boot class loader
             return;
         }
+        String className = fromCString(name);
+        if (className != null && AccessAdvisor.PROXY_CLASS_NAME_PATTERN.matcher(className).matches()) {
```

Comment:
I think it would be good to introduce a helper method for that check to ensure that it is done consistently (you use the same check in a few places).